hack/terraform-eks.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: terraform enviroment variables need to be set
+AWS_CLI="aws"
+TERRAFORM="terraform"
+KUBECTL="kubectl"
+AWS_IAM="aws-iam-authenticator"
+RUN_PATH="$HOME/tmp/terraform-eks"
+CONFIG_PATH=$(dirname $(readlink -f "$0"))/terraform/eks
+CMD_ARRAY=($AWS_CLI $TERRAFORM $KUBECTL $AWS_IAM pv bzip2 jq readlink)
```

Comment:
you check for `readlink` existence after using it so that makes no sense

Reply:
the variable CONFIG_PATH is only used after all commands are validated, So even if "readlink" does not work and $(readlink -f ...) returns some garbage, CONFIG_PATH won't be used, no?

Reply:
Even if `CONFIG_PATH` is used after command validation, it is evaluated before, so you get some extra error messages.

But I'm not really worried about `readlink` not being installed (unlike say `jq`). I am more concerned about having the code work on macOS, thus my suggestion in https://github.com/vmware-tanzu/antrea/pull/466#discussion_r394591605

Reply:
CONFIG_PATH is invoked after validation.